frontend/src/components/assessments/assessment-creation/BasicInformation.tsx

```diff
@@ -17,70 +19,90 @@ import {
+  control: Control<TestRequest, any>;
   errors: Partial<FieldErrorsImpl<{ [x: string]: any }>>;
-  validSubmit: boolean;
+  errorMessage: string;
 }
 
 const BasicInformation = ({
-  setAssessmentName,
+  setName,
   register,
+  setValue,
+  watch,
   control,
   errors,
-  validSubmit,
+  errorMessage,
 }: BasicInformationProps): React.ReactElement => {
+  const handleGradeChange = (
+    option: SingleValue<{ value: Grade; label: string }>,
+  ) => {
+    if (option) {
+      setValue("grade", option.value);
+    }
+  };
+
+  const handleCountryChange = (
+    option: SingleValue<{ value: string; label: string }>,
+  ) => {
+    if (option) {
+      setValue("curriculumCountry", option.value);
+    }
+  };
+
   const countryOptions = React.useMemo(() => countryList().getData(), []);
 
   return (
     <Box width="100%">
       <VStack align="left" spacing={8} width="75%">
+        {errorMessage && <ErrorToast errorMessage={errorMessage} />}
         <Text textStyle="eyebrow">Basic Information</Text>
-
-        {!validSubmit && (
-          <ErrorToast errorMessage="Please resolve all issues before publishing or saving" />
-        )}
-
-        <FormControl isInvalid={Boolean(errors.assessmentName)} isRequired>
+        <FormControl isInvalid={Boolean(errors.name)} isRequired>
           <FormLabel color="grey.400">Assessment Name</FormLabel>
           <Input
             placeholder="e.g. Ontario Grade 5 Pre-Term Assessment"
-            {...register("assessmentName", {
+            {...register("name", {
               onChange: (e) => {
-                setAssessmentName(e.target.value);
+                setName(e.target.value);
               },
               required: "Please enter a name for the assessment",
             })}
           />
-          <FormErrorMessage>{errors.assessmentName?.message}</FormErrorMessage>
+          <FormErrorMessage>{errors.name?.message}</FormErrorMessage>
         </FormControl>
 
         <Box width="50%">
           <Controller
             control={control}
             name="grade"
-            render={({
-              field: { onChange, value, name },
-              fieldState: { error },
-            }) => (
+            render={({ field: { name }, fieldState: { error } }) => (
               <FormControl isInvalid={Boolean(error)} isRequired>
                 <FormLabel color="grey.400">Grade Level</FormLabel>
                 <Select
                   name={name}
-                  onChange={onChange}
+                  onChange={handleGradeChange}
                   options={gradeOptions}
-                  placeholder=""
+                  placeholder="Select a grade"
+                  selectedOptionStyle="check"
                   useBasicStyles
-                  value={value}
+                  value={
+                    gradeOptions.find(
+                      (option) => option.value === watch("grade"),
+                    ) || undefined
+                  }
```

Comment:
I think we should probably split some of these `Select` components into separate components. I think there's a couple bugs related to dropdowns and also a ticket for refactoring some stuff into common components, so I think I'll work on all of these things together in a subsequent PR! So leaving in this component for now